crates/wasmtime/src/runtime/vm/instance.rs

```diff
@@ -573,9 +573,13 @@ impl Instance {
     /// # Panics
     ///
     /// Panics if `index` is out of bounds for this instance.
-    pub fn get_exported_func(self: Pin<&mut Self>, index: FuncIndex) -> ExportFunction {
+    pub fn get_exported_func(
+        self: Pin<&mut Self>,
+        store: StoreId,
+        index: FuncIndex,
+    ) -> crate::Func {
         let func_ref = self.get_func_ref(index).unwrap();
-        ExportFunction { func_ref }
+        unsafe { crate::Func::from_vm_func_ref(store, func_ref) }
```

Comment:
Can we actually encapsulate the unsafety here? Because `from_vm_func_ref` requires that the store owns the given funcref, but because we take the store as an argument but get the funcref from `self`, we don't technically have the guarantee that the given store owns this funcref. So I think either this function should be `unsafe` because it relies on passing the correct `StoreId` in, or we need to get that `StoreId` from `self` somehow.

Reply:
Good point yeah, I meant to come back and fix this up but forgot to do so. It's a bit unfortunate how infectious the unsafety is here but I agree it's required to be sound regardless.